server/gitrest/src/utils/nodegitManager.ts

```diff
@@ -331,16 +338,109 @@ export class NodegitRepositoryManager implements IRepositoryManager {
         );
         return conversions.tagToITag(await nodegit.Tag.lookup(this.repo, tagOid));
     }
+
+    public async getSummary(
```

Comment:
Yeah, I wrestled with this for a while. Really appreciate the suggestions. My main concern was that the summary persistence relies on repoPath information that does not exist outside of the scope of the RepositoryManager class, and would actually potentially be dependent on the implementation. But I think that reasoning was likely flawed, because repoPath should probably be implementation agnostic.